broker/src/bin/neon_broker.rs

```diff
@@ -0,0 +1,563 @@
+//! Simple pub-sub based on grpc (tonic) and Tokio mpsc for storage nodes
+//! messaging. The main design goal is to avoid central synchronization during
+//! normal flow, resorting to it only when pub/sub change happens. Each
+//! subscriber holds mpsc for messages it sits on; tx end is sent to existing
+//! publishers and saved in shared state for new ones. Publishers maintain
+//! locally set of subscribers they stream messages to.
+//!
+//! Subscriptions to 1) single timeline 2) everything are possible. We could add
+//! subscription to set of timelines to save grpc streams, but testing shows
+//! many individual streams is also ok.
+//!
+//! Message is dropped if subscriber can't consume it, not affecting other
+//! subscribers.
+//!
+//! Only safekeeper message is supported, but it is not hard to add something
+//! else with templating.
```

Comment:
I mean pub-sub code is agnostic to the carried message (`SafekeeperTimelineInfo` here), it is easy to extract it to template arg if we have another (e.g. similar thing for pageservers). Or alternatively we can use protobuf Any, as Arthur suggested.

Reply:
Do you mean generics by templating?

Reply:
yes

Reply:
Lets use proper terminology then. Its not c++ :). 

As a third option we can also just store message as bytes and let the client decide what needs to be passed here. It can be another protobuf blob, can be json.

Reply:
changed to generics